DqtApi/src/DqtApi/DataStore/Crm/DataverseAdapter.CreateTeacher.cs

```diff
@@ -38,14 +38,23 @@ public async Task<CreateTeacherResult> CreateTeacher(CreateTeacherCommand comman
                 return (CreateTeacherResult.Failed(failedReasons), null);
             }
 
+            var newContact = helper.CreateContactEntity();
+            string trn = null;
+            bool isUseTrnGenerationApiFeatureEnabled = await _featureManager.IsEnabledAsync(FeatureFlags.UseTrnGenerationApi);
+            if (isUseTrnGenerationApiFeatureEnabled)
+            {
+                trn = await _trnGenerationApiClient.GenerateTrnAsync();
```

Comment:
This needs to move further down and only be called when `allocateTrn` is `false` (i.e. we haven't found a duplicate).

Reply:
Moved things around